Lib/urllib/parse.py

```diff
@@ -511,23 +547,26 @@ def urlunsplit(components):
         url = url + '#' + fragment
     return _coerce_result(url)
 
-def urljoin(base, url, allow_fragments=True):
+def urljoin(base, url, allow_fragments=True, flags=SchemeFlag(0)):
     """Join a base URL and a possibly relative URL to form an absolute
-    interpretation of the latter."""
+    interpretation of the latter. Some logic may be enabled by setting
+    the classes variable."""
```

Comment:
s/classes/flags/

Reply:
I also suggest making the new parameter be keyword only.